pkg/operator/ceph/object/health.go

```diff
@@ -0,0 +1,272 @@
+		7. PUT the file
+		8. GET the file
+		9. compare MD5
+		10. delete object on bucket
+		11. update CR health status check
+
+		Always keep the bucket and the user for the health check, just do PUT and GET because bucket creation is expensive
+	*/
+
+	var s3AccessKey string
+	var s3SecretKey string
+	s3endpoint := fmt.Sprintf("%s:%s", BuildDomainName(c.objContext.Name, c.namespacedName.Namespace), c.port)
+
+	// Generate unique user and bucket name
+	bucket := genUniqueBucketName(c.objContext.UID)
+	userConfig := c.genUserConfig()
+
+	// Create S3 user
+	logger.Debugf("creating s3 user object %q for object store %q", userConfig.UserID, c.namespacedName.Name)
+	user, rgwerr, err := CreateUser(c.objContext, userConfig)
+	if err != nil {
+		if rgwerr == ErrorCodeFileExists {
+			user, _, err = GetUser(c.objContext, userConfig.UserID)
+			if err != nil {
+				err = errors.Wrapf(err, "failed to get details from ceph object user %q for object store %q", user.UserID, c.namespacedName.Name)
+				updateStatusEndpoint(c.client, c.namespacedName, cephv1.ConditionFailure, err.Error())
+				return err
+			}
+		} else {
+			err = errors.Wrapf(err, "failed to create object user %q. error code %d for object store %q", userConfig.UserID, rgwerr, c.namespacedName.Name)
+			updateStatusEndpoint(c.client, c.namespacedName, cephv1.ConditionFailure, err.Error())
+			return err
+		}
+	}
+	// Set access and secret key
+	s3AccessKey = *user.AccessKey
+	s3SecretKey = *user.SecretKey
+
+	// Initiate s3 agent
+	logger.Debugf("initializing s3 connection for object store %q", c.namespacedName.Name)
+	s3client := CreateNewS3Client(s3endpoint, s3AccessKey, s3SecretKey)
+
+	// Force purge the s3 object before starting anything
+	err = cleanupObjectHealthCheck(s3client, c.objContext.UID)
+	if err != nil {
+		err = errors.Wrapf(err, "failed to perform object cleanup for object store %q", c.namespacedName.Name)
+		updateStatusEndpoint(c.client, c.namespacedName, cephv1.ConditionFailure, err.Error())
+		return err
+	}
+
+	// Bucket health test
+	err = c.testBucketHealth(s3client, bucket)
+	if err != nil {
+		err = errors.Wrapf(err, "failed to run bucket health checks for object store %q", c.namespacedName.Name)
+		updateStatusEndpoint(c.client, c.namespacedName, cephv1.ConditionFailure, err.Error())
+		return err
+	}
+
+	logger.Debug("successfully checked object store endpoint for object store %q", c.namespacedName.Name)
+	// Update the EndpointStatus in the CR to reflect the healthyness
+	updateStatusEndpoint(c.client, c.namespacedName, cephv1.ConditionConnected, "")
+
+	return nil
+}
+
+func cleanupObjectHealthCheck(s3client *S3Client, objectStoreUID string) error {
+	bucketToDelete := genUniqueBucketName(objectStoreUID)
+	logger.Infof("deleting object %q from bucket %q", s3HealthCheckObjectKey, bucketToDelete)
+	_, err := s3client.DeleteObjectInBucket(bucketToDelete, s3HealthCheckObjectKey)
+
+	return err
+}
+
+func (c *bucketChecker) cleanupHealthCheck() error {
```

Comment:
Since it's best effort, no need for the return err
```suggestion
func (c *bucketChecker) cleanupHealthCheck() {
```